class.jetpack-options.php

```diff
@@ -153,13 +183,28 @@ public static function get_option( $name, $default = false ) {
 	 * @param string $name Option name
 	 * @param mixed $default (optional)
 	 *
-	 * @return mixed|void
+	 * @return mixed
 	 */
 	public static function get_option_and_ensure_autoload( $name, $default ) {
-		$value = get_option( $name );
+		// In this function the name is not adjusted by prefixing jetpack_
+		// so if it has already prefixed, we'll replace it and then
+		// check if the option name is a network option or not
+		$jetpack_name = preg_replace( '/^jetpack_/', '', $name, 1 );
+		$is_network_option = self::is_network_option( $jetpack_name );
+		if ( $is_network_option ) {
+			$value = get_site_option( $name );
+		}
+		else {
```

Comment:
Core coding style should have this as `} else {` but I could see it as a ternary.  Either way.